R/util-checkSpec.R

```diff
@@ -54,6 +54,30 @@ CheckSpec <- function(lData, lSpec) {
   allCols <- c()
   missingCols <- c()
   for (strDataFrame in lSpecDataFrames) {
+    chrDataFrameColnames <- colnames(lData[[strDataFrame]])
+    #check modes in data
+    wrongType <- purrr::reduce2(
+      lSpec[[strDataFrame]],
+      names(lSpec[[strDataFrame]]),
+      function(so_far, x, idx) {
+        if (!is.null(x$type) && idx %in% chrDataFrameColnames) {
+          #check if data is the expected mode
+          res <- all(x$type %in% mode(lData[[strDataFrame]][[idx]]))
+          if (!res) {
+            so_far <- c(so_far, idx)
+          }
+          return(so_far)
+        }
+      },
+      .init = character()
+    )
+
+    if (length(wrongType)) {
+      cli::cli_alert_danger("Not all columns of {strDataFrame} in the spec are in the expected format, improperly formatted columns are: {wrongType}")
```

Comment:
I really feel like these should be `cli::cli_abort()` (see #1678), so we don't keep processing when we already know that it fails the check (`cli_alert_danger()` tells the user that something is wrong but doesn't stop the process). If we did that, we could/should put the required column checks before the type checks.

Reply:
yeah, i think this is right. I had this thought as I was writing it, but chose to just match the existing messages instead of challenging status quo🤦‍♀️. so to align with this, we'll need to update all of the existing cli messages in this function but i think i'll go ahead and do that.

Reply:
@lauramaxwell I think @jwildfire had thoughts about this/reasons for it to just be the `cli_alert_danger()`, so I'm not sure.

Reply:
i've noted this in #1678 and added that issue to the v2.1.0 milestone. can discuss further if needed, but i think its a separate issue from this PR

Reply:
Agreed!